### PR TITLE
update rubocop to 0.48.1

### PR DESCRIPTION
**Pull request by agrim123**
Addresses #1450

### CODE REVIEW

**Comment by agrim123** [comment]
@Ana06 On running `bundle update rubocop` I got the error for `nokogiri`. So I ran `bundle update nokogiri` but it downgraded the version to `1.6.8.1` and that's why hakiri is generating warnings.

**Comment by differentreality** [comment]
Don't you think we should keep nokogiri at a version (>=1.7.1 according to Hakiri) for which hakiri doesn't complain, even if that means we update more gems?

**Comment by agrim123** [comment]
But then I have to manually modify Gemfile.lock cause each time I run bundle update nokogiri it downgrades the gem! And without it, I am unable to update rubocop!

**Comment by shlok007** [comment]
Have a look at https://github.com/openSUSE/osem/pull/1396#issuecomment-293786809

>But then I have to manually modify Gemfile.lock 'cause each time I run bundle update nokogiri it downgrades the gem! And without it, I am unable to update rubocop!

Have a look at these http://bundler.io/man/bundle-update.1.html#PATCH-LEVEL-OPTIONS  and see if any one of that helps.

**Comment by Ana06** [comment]
@agrim123 the way to update a gem is:

1. Change the version in the [Gemfile](https://github.com/openSUSE/osem/blob/master/Gemfile#L202), in this case:
`gem 'rubocop', '~> 0.48.1', require: false`

2. Run `bundle install`
This way the dependencies will be automatically solved.

And again **you shouldn't manually modify the Gemfile.lock**

**Comment by Ana06** [comment]
@agrim123 remember you need to generate the `.rubocop_todo.yml` file :wink:

**Comment by agrim123** [comment]
@Ana06 I already generated it using
`rubocop --auto-gen-config`

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -323,6 +323,11 @@ Style/ConditionalAssignment:
     - 'app/helpers/application_helper.rb'
     - 'db/migrate/20140610165551_migrate_data_person_to_user.rb'
     - 'db/migrate/20140820124117_undo_wrong_migration20140801080705_add_users_to_events.rb'
+    - 'app/models/ticket_purchase.rb'
```
Comment (agrim123) [inline .rubocop_todo.yml]: done :+1:

**Comment by differentreality** [comment]
Please rebase

**Comment by agrim123** [comment]
@Ana06 
This is the original file that exists in `upstream` 
https://github.com/openSUSE/osem/blob/master/.rubocop_todo.yml#L199-L214

And in my second [commit](https://github.com/openSUSE/osem/pull/1451/commits/5a184c513950915a5e5c2c00d42cd7fe575b2dfa), I have regenerated the file and those deletions are because of that.
In my third [commit](https://github.com/openSUSE/osem/pull/1451/commits/16adc447cad6c65ee466e36790ad512dca0b8234),
 I added the conflict files manually.

**Comment by Ana06** [comment]
@agrim123 
> In my third commit, I added the conflict files manually.

When rebasing you had to regenerate the todo file again, and those changes should be in the second commit, but hey are in the third. The only cop affected by the Rubocop bug is `ConditionalAssignment`, and you are changing more things in the third commit. Those things shouldn't be there.

**Pull request merged**